functional/FunctionalTest.java

```diff
@@ -1033,12 +1033,24 @@ public static void getObject_test3() throws Exception {
     long startTime = System.currentTimeMillis();
     try {
       String objectName = getRandomName();
-      try (final InputStream is = new ContentInputStream(3 * MB)) {
-        client.putObject(bucketName, objectName, is, 3 * MB, nullContentType);
+      try (final InputStream is = new ContentInputStream(1024)) {
```

Comment:
Add new tests than changing existing ones.

Reply:
I've moved the new test case (with the assertions about the bytes that should be received from the partial getObject request) into `getObject_test8` in the functional suite. I had to make minor updates to two other functional tests that were leaving behind objects in the shared bucket and causing `listObject_test5` to fail in error; those other tests should now be cleaning up after themselves.